node/core/av-store/src/lib.rs

```diff
@@ -1177,7 +1178,7 @@ fn store_available_data(
 	let erasure_chunks = chunks.iter().zip(branches.map(|(proof, _)| proof)).enumerate().map(
 		|(index, (chunk, proof))| ErasureChunk {
 			chunk: chunk.clone(),
-			proof,
+			proof: Proof::try_from(proof).unwrap(),
```

Comment:
Note: I am not sure what the best approach is here.

Since the `obtain_chunks` function (c.f. [here](https://github.com/paritytech/polkadot/blob/master/erasure-coding/src/lib.rs#L136)) and `branches` function (c.f. [here](https://github.com/paritytech/polkadot/blob/686e3180af8fee86b2b541ffb9b09d7bc66eba9c/erasure-coding/src/lib.rs#L276).

One option would be to have the erasure-coding return the Proof as oppose to the `AsRef<[u8]>` and the `Vec<Vec<u8>>`. But before I go ahead and do that it maybe be best to sync up regarding the proper approach.